### ISSUE TITLE
Meet requirements for deferred Clippy warnings

### ISSUE THREAD

**Issue by marcelstanley** [cartesi/rollups-node#37]
## 📚 Context

As first mentioned at cartesi/rollups#159, Clippy raises a few warnings which could not be addressed at said PR.

In the meantime, the corresponding lints listed below were disabled during CI runs:

- [*new_ret_no_self*](https://rust-lang.github.io/rust-clippy/master/index.html#/new_ret_no_self);
- [*module_inception*](https://rust-lang.github.io/rust-clippy/master/index.html#/module_inception) - affects code generated via `state-fold-types`.

## ✔️ Solution

For, *module_inception*, we'd have the choice of suppressing the warnings via configuration or changing the business logic to meet the lint requirements. Both options would depend on cartesi/rollups-node#31 being completed.

In the case of *new_ret_no_self*, we would need changing the business logic to return the proper types.

## 📈 Subtasks

Meet requirements for:

- [x] *[upper_case_acronyms](https://rust-lang.github.io/rust-clippy/master/index.html#upper_case_acronyms)*, allowed at [02f79f9](https://github.com/cartesi/rollups/commit/02f79f93eb9c4907793bd49a5ec8aa6849d3097e)
- [x] *new_ret_no_self*
- [x]  ~*too_many_arguments*~, disabled at [93ac09](https://github.com/cartesi/rollups/commit/93ac098f3760d636228c70bfcf6d6c4b8aec6486)
- [x] ~*module_inception*~ or suppress it via linter annotations (depends on #31)

**Comment by marcelstanley** [comment]
Let's postpone fixing the warning for too_many_arguments](https://github.com/cartesi/rollups/commit/93ac098f3760d636228c70bfcf6d6c4b8aec6486) for when the Dispatcher is rewritten.

**Comment by marcelstanley** [comment]
Also, *module_inception* should be dealt with when the state-fold is refactored after it's merged as part of #31.

**Comment by marcelstanley** [comment]
Remaining warnings to be fixed at #46
